lib/puppet/network/http/rack/rest.rb

```diff
@@ -103,7 +103,21 @@ def client_cert(request)
     # NOTE: The SSL_CLIENT_CERT environment variable will be the empty string
     # when Puppet agent nodes have not yet obtained a signed certificate.
     if cert.nil? || cert.empty?
-      nil
+      # When running with unicorn, the SSL_CLIENT_CERT variable is not available
+      # in the environment, therefore we have to pass a header: 'X-SSL-Client-Cert'
+      cert = request.env['HTTP_X_SSL_CLIENT_CERT']
```

Comment:
On second review this is a header and not a param, and Rack might not directly expose headers in a separate object and only provides them in the request environment. If it's possible to extract this value as a header I would prefer that, but if this the best way to do this then I'm fine with it.